tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_configure_backend_equivalency.py

```diff
@@ -0,0 +1,151 @@
+"""Tests related to the equivalency feature of `configure_backend`."""
+
+from pathlib import Path
+from typing import Literal, Tuple
+
+import numcodecs
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+from pynwb import NWBFile, ProcessingModule
+from pynwb.testing.mock.base import mock_TimeSeries
+from pynwb.testing.mock.file import mock_NWBFile
+
+from neuroconv.tools.nwb_helpers import (
+    BACKEND_NWB_IO,
+    configure_backend,
+    get_default_backend_configuration,
+    get_module,
+)
+
+
+def _generate_integer_array(
+    seed: int,
+    dtype: np.dtype = np.dtype("int16"),
+    shape: Tuple[int, int] = (30_000 * 5, 384),
+) -> np.ndarray:
+    random_number_generator = np.random.default_rng(seed=seed)
+
+    low = np.iinfo(dtype).min
+    high = np.iinfo(dtype).max
+    data = random_number_generator.integers(low=low, high=high, size=shape, dtype=dtype)
+
+    return data
+
+
+@pytest.fixture(scope="module")
+def array_1(seed: int = 0) -> NWBFile:
+    integer_array = _generate_integer_array(seed=seed)
+
+    return integer_array
+
+
+@pytest.fixture(scope="module")
+def array_2(seed: int = 1) -> NWBFile:
+    integer_array = _generate_integer_array(seed=seed)
+
+    return integer_array
+
+
+@pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+def test_configure_backend_equivalency(
+    tmpdir: Path,
+    array_1: NWBFile,
```

Comment:
Is the type of `array_` NWBFile? Just double checking?

Reply:
Oops

So I started off attempting to use `nwbfile`'s as fixtures, which you might notice would have reduced a lot of duplication in places

But they had mutation issues that caused really weird issues (individual tests passed if you ran them in isolation but not the entire suite after the first passing check)

Reply:
Though maybe I just need to try again with `scope="function"`...

Reply:
Nah, I think that provides too much redirection - it's useful to see the slightly different `NWBFile` structures laid out side by side but the irrelevant detail of the integer array abstracted to the top

Reply:
Could be, I remember having the same type of trouble when using fixtures in the spikeinterface tests.